src/core/infer_request.cc

```diff
@@ -116,9 +243,43 @@ InferenceRequest::Release(std::unique_ptr<InferenceRequest>&& request)
 InferenceRequest*
 InferenceRequest::CopyAsNull(const InferenceRequest& from)
 {
-  // FIXME
-  LOG_ERROR << "CopyAsNull: NYI";
-  return nullptr;
+  // Create a copy of 'from' request with artifical inputs and no requested
+  // outputs. Maybe more efficient to share inputs and other metadata,
+  // but that binds the Null request with 'from' request's lifecycle.
+  std::unique_ptr<InferenceRequest> lrequest(
+      new InferenceRequest(from.backend_raw_, from.requested_model_version_));
+  lrequest->needs_normalization_ = false;
+  lrequest->batch_size_ = from.batch_size_;
+
+  // Two pass: first to obtain the max input byte size for allocating a large
+  // enough buffer for all inputs; second to construct the inputs
+  size_t max_byte_size = 0;
+  for (const auto& input : from.OriginalInputs()) {
+    if (input.second.Data()->TotalByteSize() > max_byte_size) {
+      max_byte_size = input.second.Data()->TotalByteSize();
+    }
+  }
+
+  auto data_sp = persistent_buffer_manager->GetBuffer(max_byte_size);
+  for (const auto& input : from.OriginalInputs()) {
+    Input* new_input;
+    lrequest->AddOriginalInput(
+        input.first, input.second.DType(), input.second.Shape(), &new_input);
+    new_input->AppendData(
+        data_sp->buffer_, input.second.Data()->TotalByteSize(),
+        TRITONSERVER_MEMORY_CPU, 0);
+  }
+
+  lrequest->SetResponseCallback(
+      &null_allocator, nullptr, NullResponseComplete, nullptr);
```

Comment:
Add a comment saying that since no outputs were requested there should be no allocations.